### PR TITLE
Feature/post functionality 2

### PR DESCRIPTION
**Pull request by JeffShepherd**
## What does this PR do?

- Adds post functionality. A user can input today's data, and upon clicking submit, the data will be posted to the relevant API endpoint.

## How should it be tested in the UI?

- run npm start in the app and api.
- for each category
  - fill in the form and hit submit
  - check the data in the endpoint and ensure the submission appears
  - try another submission for the same category, and an error message should appear due to data existing for that day already
  - form should reset no matter what happens in the request

- Not in scope: DOM updates after fetch request is completed

### CODE REVIEW

File: src/api.js [hunk 1]
```diff
@@ -1,56 +1,67 @@
 //GET requests
 export const sleepDataAPI = fetch("http://localhost:3001/api/v1/sleep")
   .then(response => response.json())
+  .catch(err => alert('Sorry, but there was an error. Please try again.'))
```
Comment (Shakikka) [inline src/api.js:4]: I wish you kept the "Not Enough Sauce" Alert...but nice work on adding these here.

File: src/scripts.js [hunk 2]
```diff
@@ -68,37 +65,45 @@ const userForms = document.querySelectorAll('.user-form');
...
+  userRepo = new UserRepo(userInfo);
+  randomUser = pickUser(userRepo);
 
   addInfoToSidebar(randomUser, userRepo);
   displayHydrationInfo(userRepo, randomUser.id, hydrationInfo);
   displaySleepInfo(userRepo, randomUser.id, sleepInfo);
 }
 
 function displayHydrationInfo(userInfo, id, hydrationInfo) {
-  let hydrationRepo = new Hydration(hydrationInfo);
```
Comment (Shakikka) [inline src/scripts.js:100]: I think you accidentally deleted the wrong thing when merging here. I removed `makeToday()` in my last PR, as there was a redundant method in UserRepo called `.getToday()`.

File: src/scripts.js [hunk 3]
```diff
@@ -109,8 +114,9 @@ function displayHydrationInfo(userInfo, id, hydrationInfo) {
 }
 
 function displaySleepInfo(userInfo, id, sleepInfo) {
-  let sleepRepo = new Sleep(sleepInfo);
-  let today = userInfo.getToday(id, sleepRepo.sleepData);
```
Comment (Shakikka) [inline src/scripts.js:113]: Same as above

**Review by Shakikka — CHANGES_REQUESTED** [review]
Nice work on this Jeff. The Activity data and Hydration data are posting as expected, however the Sleep data is throwing an error, I think due to an accidental deletion as described in the comments for scripts.js.

File: src/scripts.js (hunk 2 above)
Comment (JeffShepherd) [reply src/scripts.js:100]: fixed and pushed up

File: src/scripts.js (hunk 3 above)
Comment (JeffShepherd) [reply src/scripts.js:113]: fixed and pushed up

**Review by Shakikka — APPROVED** [review]

**Pull request merged**